crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2272,46 +2270,13 @@ impl<'db> TypeInferenceBuilder<'db> {
                 .ok_or(ModuleNameResolutionError::InvalidSyntax)
         };
 
-        let ty = match module_name {
-            Ok(module_name) => {
-                if let Some(module_ty) = self.module_ty_from_name(&module_name) {
-                    let ast::Alias {
-                        range: _,
-                        name,
-                        asname: _,
-                    } = alias;
```

Comment:
I took the liberty of reorganizing this logic to use early-return to tease apart the different cases.  I find it easier to reason about this way, but I'm happy to revert back to the previous style if that's house style.  Other than the part I call out below, most of this diff should just be a refactoring.

Reply:
I think we (or at least I) have tended towards expression style rather than early return style, probably particularly in this case to avoid the repeat calls to `add_declaration_with_binding`. But I don't really have a strong preference for that, and I agree that in this case your version approves readability. In other words: feel free to improve our house style, that's what you're here for :)

Reply:
Just some additional data point: Early returns are used extensively in Ruff.